security/fma-kona-asterisc.md

```diff
@@ -0,0 +1,213 @@
+  Once `kona` + `asterisc` is the respected game type, the dispute monitor will watch for all new games and will alert if there is an issue with `kona` + `asterisc`.
+  If `kona` + `asterisc` has an issue while in the hot path, and an invalid output / game is played, the dispute monitor and `op-challenger` would alert and detect invalid games.
+  At this point, the permissioned fallback would need to be engaged since both proof systems are in repair.
+
+- **Recovery Path(s):**
+  Likely the only viable recovery path would be the permissioned fallback unless the `op-program` + `cannon` are fixed and able to be placed back into the hotpath.
+
+
+### Divergent Kona Derivation Pipeline
+
+- **Description:**
+  The `kona` derivation pipeline diverged in behavior from the `op-program`'s derivation pipeline.
+
+- **Risk Assessment:**
+  Low likelihood, low impact.
+  Low likelihood as the derivation pipeline is extensively unit tested, run in the VM Runner, and is action tested.
+  Low impact since by default `kona` + `asterisc` are placed in the cold path.
+  [The "Kona and Asterisc Hot Path Failure"](#Kona-and-Asterisc-Hot-Path-Failure) above covers the failure mode for `kona` + `asterisc` when placed in the hot path.
+
+- **Mitigations:**
+  Action tests covering the derivation pipeline are run on every pull request in the [kona] repository.
+  VM Runner executes the derivation pipeline as part of the kona fault proof program running on top of asterisc.
+  [Extensive test coverage](https://app.codecov.io/gh/anton-rs/kona/tree/main/crates) of kona.
+
+- **Detection:**
+  The VM runner would pick up an issue.
+  The action tests or local testing may pick up an issue.
+  Use of the derivation elsewhere, for example in alternative consensus clients like `op-rs`, would pick up an issue.
+
+- **Recovery Path(s):**
+  Fix the derivation pipeline divergence, release and deploy the updated kona + asterisc when in the coldpath.
+  If `kona` + `asterisc` is placed in the hotpath, the system operator would either need to fall back to `op-program` and `cannon`, or fall back to the permissioned dispute game.
+  [The "Kona and Asterisc Hot Path Failure"](#Kona-and-Asterisc-Hot-Path-Failure) above covers the failure mode for `kona` + `asterisc` when placed in the hot path.
+
+
+### Kona Execution Diverges from op-program
+
+- **Description:**
+  The Kona Fault Proof Program executes blocks in a stateless L2 block executor that uses optimism-revm as a backend.
+  If optimism-revm or the L2 block executor diverge in behavior from the `op-program`, the `kona` + `asterisc` proof system will be invalid.
+
+- **Risk Assessment:**
+  Low likelihood, low impact.
+  A small divergence in execution behavior can cause invalid game results.
+  But since kona + asterisc is in the coldpath, it doesn't have impact on anything in the production system.
+
+- **Mitigations:**
+  State testing against op-revm.
```

Comment:
For the various detections and mitigations listed in this doc, "state testing", "action tests", op-rs, monitoring, etc, we should either:
- Link out to, or name, the specific tests used, or
- Track the implementation of that mitigation in the Action Items section

It looks like currently there are no action items which would imply all detections and mitigations are implemented

Reply:
Links have been added, let's just added the standard Action Items header and initial item that way we know we considered actions even if we have none because they are already done

Reply:
Done